internal/provider/resource_password.go

```diff
@@ -192,8 +199,20 @@ func upgradePasswordStateV0toV2(ctx context.Context, req resource.UpgradeStateRe
 		return
 	}
 
-	passwordDataV2 := passwordModelV2{
-		Keepers:         passwordDataV0.Keepers,
+	keepers := types.Map{Null: true, ElemType: types.StringType}
+
+	if len(passwordDataV0.Keepers.Elems) > 0 {
+		keepers = passwordDataV0.Keepers
+	}
+
+	overrideSpecial := types.String{Null: true}
+
+	if len(passwordDataV0.OverrideSpecial.Value) > 0 {
+		overrideSpecial = passwordDataV0.OverrideSpecial
+	}
```

Comment:
I _want_ to do this but I believe it would unfortunately cause practitioner plan differences if their configuration did happen to be:

```go
resource "random_password" "example" {
  # ... other attributes ...
  override_special = ""
}
```

Which is perfectly valid and may be coming from something like a module variable declaration:

```terraform
variable "password_special_characters" {
  default     = ""
  description = "Set this to override the special characters that can be included in the password. Defaults to !@#$%&*()-_=+[]{}<>:?"
  type        = string
}

resource "random_password" "example" {
  # ... other attributes ...
  override_special = var.password_special_characters
}
```

For example, the next plan after state upgrade I believe Terraform would show the below (or potentially would show that there's an in-place update, but not specifically this line because it might be silenced due to legacy SDK plan handling):

```
# resource random_password.test will be updated in-place
~ resource "random_password" "test" {
  ~ override_special: null => ""
}
```

Reply:
You are correct. Adding a state upgrader to do the following:

```go
	keepers := types.Map{Null: true, ElemType: types.StringType}

	if len(passwordDataV2.Keepers.Elems) > 0 {
		keepers = passwordDataV2.Keepers
	}

	overrideSpecial := types.String{Null: true}

	if len(passwordDataV2.OverrideSpecial.Value) > 0 {
		overrideSpecial = passwordDataV2.OverrideSpecial
	}
```

Will result in practitioner plan differences if they used `"override_special": ""` and/or `"keepers": {}` in their configuration.

If `"override_special": ""` is used in the config to create the resource (e.g., using `v3.4.2`) and then _terraform apply_ is run with this branch the following is seen:
```
  # random_password.test must be replaced
-/+ resource "random_password" "test" {
      ~ bcrypt_hash = (sensitive value)
      ~ id          = "none" -> (known after apply)
      ~ result      = (sensitive value)
        # (10 unchanged attributes hidden)
    }
```


If `"keepers": {}` is used in the config to create the resource (e.g., using `v3.4.2`) and then _terraform apply_ is run with this branch the following is seen:
```
  # random_password.test will be updated in-place
  ~ resource "random_password" "test" {
        id          = "none"
      + keepers     = {}
        # (12 unchanged attributes hidden)
    }
```